DEVELOPER_GUIDE.md

```diff
@@ -1,29 +1,44 @@
 
-Those _Type Class Implementation Function Types_ define static methods of several kinds:
+The concrete implementation of these abstractions on the other hand is done, minimally, by defining a concrete type (an Invokable) per function.
+
+For example, one of the simplest abstractions `Monoid` has its core implementation split under two types `Plus` and `Zero`.
+
+Those Invokers contains static methods of several kinds:
+
+* concrete overloads, for example Monoid's `Plus` will have overloads of a `Plus` method (named the same as the type defining those) for each of the BCL and FSharp.Core types that ought to support the `Monoid` type class, e.g. one to concatenate two strings, one to concatenate two lists, one to concatenate two arrays, and so on and so forth.
+* an _invokable_ method, that should always be defined as `inline`, which embeds the SRTP dispatcher machinery that is used to pick the right overload (whether defined in F#+ or outside of it)
+* a default or set of default implementations, those may need to be defined in an instrinsic type extension due to F# compiler implementation details to enable being picked up correctly for types that end up supporting the abstraction when such abstraction comes with a default implementation expressed in terms of composing several functions, those are always defined as `inline`.
 
-* concrete overloads, for exemple Monoid's `Plus` will have overloads of a `Plus` method (named the same as the type defining those) for each of the BCL and FSharp.Core types that ought to support the `Monoid` type class, e.g. one to concatenate two strings, one to concatenate two lists, one to concatenate two arrays, and so on and so forth.
-* an _invoker_ method, that should always be defined as `inline`, which embeds the SRTP machinery that is used to pick the right overload (whether defined in F#+ or outside of it for implementing type class)
-* a default or set of default implementations, those may need to be defined in a type extension due to F# compiler implementation details to enable being picked up correctly for types that end up supporting the type class when such type class come with default implementation expressed in terms of composing several functions, those are also always defined as `inline`.
 
-### Implementation idioms
 
 #### Concrete implementations
 
+We should add as much concrete implementations for primitive types as possible, since the end user won't be able to add them later.
+
 #### Defaults
 
+The goal of defaults is to allow users of the library to write less code, but their are not intended to be used by developers of F#+ as we should afford writing more code in order to maximize usability of the library.
```

Comment:
We describe a goal of defaults, but it's not clear what are they. Do you mean "a default or set of default implementations" which were described above?
I'm also confused about how they are not intended to be used by developers?
Or do we want to say that we try to make a library which principle is to allow user to write less boilerplate, and because of that we have to put some boilerplate inside F#+?

Reply:
Yes, it's rather  the latter and it's not clear at all, I wrote it this way mainly for me to not forget to include this idea in the guideline, but need to come back later and do the effort of explaining it in a better way, ideally with some concrete examples.